src/ring_theory/dedekind_domain.lean

```diff
@@ -108,33 +108,150 @@ structure is_dedekind_domain_dvr : Prop :=
 (is_dvr_at_nonzero_prime : ∀ P ≠ (⊥ : ideal A), P.is_prime →
   discrete_valuation_ring (localization.at_prime P))
 
+section inverse
+
+open_locale classical
+
+variables {R₁ : Type*} [integral_domain R₁] {g : fraction_map R₁ K}
+
+noncomputable instance fractional_ideal_has_div :
+  has_div (fractional_ideal g) :=
+⟨ λ I J, if h : J = 0 then 0 else ⟨I.1 / J.1, fractional_ideal.fractional_div_of_nonzero h⟩ ⟩
+
+variables {I J : fractional_ideal g} [fact (J ≠ 0)]
+
+noncomputable instance : has_inv (fractional_ideal g) := ⟨λ I, 1 / I⟩
```

Comment:
I'm confused. I thought you didn't want this instance in general, only for dedekind domains... But this instance is for arbitrary integral domains, right?

Reply:
Well, the point discussed with Anne was that we'd rather start by moving all `inv`-related stuff inside `dedekind_domain`, removing it from `fractional_ideal` to prevent "inverses of non-invertible ideals`. We will then try to come up with the best solution for `dedekind_domains`, either via checking that all ideals are a `group_with_0` or via this instance (but defining `is_invertible` for fractional ideals and proving that for a Dedekind domain this is always satisfied).